framework/helpers/BaseHtml.php

```diff
@@ -828,10 +828,6 @@ public static function listBox($name, $selection = null, $items = [], $options =
         }
         $options['name'] = $name;
         if (isset($options['unselect'])) {
-            // add a hidden field so that if the list box has no option being selected, it still submits a value
-            if (!empty($name) && substr_compare($name, '[]', -2, 2) === 0) {
```

Comment:
I'd keep this change but it might break BC. Maybe instead of removing this we should right trim brackets from query params when comparing them in _applyFilter_?